File: app/src/main/java/com/agera/hometools/utils/AppendMap.java
package com.agera.hometools.utils;

import java.util.HashMap;

/**
 * Created by  Agera K on 2017/9/22.
 */
public class AppendMap {

    private HashMap map = new HashMap();

    public AppendMap() {
        map.clear();
    }

    public AppendMap put(String key, String value) {
        map.put(key, value);
        return this;
    }

    public HashMap compile() {
        return map;
    }
}

File: app/src/main/java/com/agera/hometools/utils/AppendMap.kt
package com.agera.hometools.utils

import java.util.*

/**
 * Created by Agera on 2017/11/8.
 */
class AppendMap<T> {
    val map: HashMap<String, T> = HashMap()

    init {
        map.clear()
    }

    fun put(key: String, value: T): AppendMap<T> {
        map.put(key, value)
        return this
    }

    fun compile(): HashMap<String, T> {
        return map
    }
}